File: backend/domains/auth/http/router.go
package http

import (
	"github.com/cloudwego/hertz/pkg/route"
	"github.com/erweixin/go-genai-stack/backend/domains/auth/handlers"
)

// RegisterRoutes 注册 Auth 领域的路由
//
// 参数：
//   - r: 路由分组
//   - deps: Handler 依赖容器
//
// 路由列表：
//   - POST   /auth/register   - 用户注册（无需认证）
//   - POST   /auth/login      - 用户登录（无需认证）
//   - POST   /auth/refresh    - 刷新 Token（无需认证）
func RegisterRoutes(
	r *route.RouterGroup,
	deps *handlers.HandlerDependencies,
) {
	// Auth API 分组（无需认证）
	authGroup := r.Group("/auth")
	{
		// 用户注册
		authGroup.POST("/register", deps.RegisterHandler)

		// 用户登录
		authGroup.POST("/login", deps.LoginHandler)

		// 刷新 Token
		authGroup.POST("/refresh", deps.RefreshTokenHandler)
	}
}

File: backend-nodejs/domains/auth/http/router.ts
/**
 * Auth 路由注册
 * 注册所有 Auth 相关的 HTTP 路由
 */

import type { FastifyInstance } from 'fastify';
import type { RedisClientType } from 'redis';
import type { HandlerDependencies } from '../handlers/dependencies.js';
import type { RegisterRequest, LoginRequest, RefreshTokenRequest } from './dto/auth.js';
import {
  RegisterRequestSchema,
  LoginRequestSchema,
  RefreshTokenRequestSchema,
} from './dto/auth.js';
import { registerHandler } from '../handlers/register.handler.js';
import { loginHandler } from '../handlers/login.handler.js';
import { refreshTokenHandler } from '../handlers/refresh_token.handler.js';
import {
  createLoginRateLimitMiddleware,
  createRegisterRateLimitMiddleware,
} from '../../../infrastructure/middleware/ratelimit.js';

/**
 * 注册 Auth 路由
 *
 * @param app Fastify 实例
 * @param deps Handler 依赖
 * @param redis Redis 客户端（可选，用于限流）
 */
export function registerAuthRoutes(
  app: FastifyInstance,
  deps: HandlerDependencies,
  redis: RedisClientType | null = null
): void {
  // 创建限流中间件
  const loginRateLimit = createLoginRateLimitMiddleware(redis);
  const registerRateLimit = createRegisterRateLimitMiddleware(redis);

  // POST /api/auth/register - 用户注册（限流：每小时最多 3 次）
  app.post<{ Body: RegisterRequest }>(
    '/api/auth/register',
    {
      preHandler: [registerRateLimit],
      schema: {
        body: RegisterRequestSchema,
      },
    },
    async (req, reply) => {
      await registerHandler(deps, req, reply);
    }
  );

  // POST /api/auth/login - 用户登录（限流：每分钟最多 5 次）
  app.post<{ Body: LoginRequest }>(
    '/api/auth/login',
    {
      preHandler: [loginRateLimit],
      schema: {
        body: LoginRequestSchema,
      },
    },
    async (req, reply) => {
      await loginHandler(deps, req, reply);
    }
  );

  // POST /api/auth/refresh - 刷新 Token（不需要限流，因为需要有效的 refresh token）
  app.post<{ Body: RefreshTokenRequest }>(
    '/api/auth/refresh',
    {
      schema: {
        body: RefreshTokenRequestSchema,
      },
    },
    async (req, reply) => {
      await refreshTokenHandler(deps, req, reply);
    }
  );
}
